src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfacePromptForChoice.cs

```diff
@@ -292,47 +272,32 @@ public override int PromptForChoice(string caption, string message, Collection<C
         {
             System.Management.Automation.Diagnostics.Assert(defaultChoiceKeys != null, "defaultChoiceKeys cannot be null.");
 
-            ConsoleColor fg = RawUI.ForegroundColor;
-            ConsoleColor bg = RawUI.BackgroundColor;
-            int lineLenMax = RawUI.WindowSize.Width - 1;
-            int lineLen = 0;
-
-            const string choiceTemplate = "[{0}] {1}  ";
-
+            var prompts = new List<string>();
             for (int i = 0; i < hotkeysAndPlainLabels.GetLength(1); ++i)
             {
-                ConsoleColor cfg = PromptColor;
-                if (defaultChoiceKeys.ContainsKey(i))
-                {
-                    // Should be a skin lookup
-                    cfg = DefaultPromptColor;
-                }
+                var color = defaultChoiceKeys.ContainsKey(i)
+                    ? PSStyle.Instance.Prompt.ChoiceDefault
+                    : PSStyle.Instance.Prompt.ChoiceOther;
 
+                const string choiceTemplate = "[{0}] {1}";
                 string choice =
                     string.Format(
                         CultureInfo.InvariantCulture,
                         choiceTemplate,
                         hotkeysAndPlainLabels[0, i],
                         hotkeysAndPlainLabels[1, i]);
-                WriteChoiceHelper(choice, cfg, bg, ref lineLen, lineLenMax);
-                if (shouldEmulateForMultipleChoiceSelection)
-                {
-                    WriteLineToConsole();
-                }
-            }
 
-            WriteChoiceHelper(
-                ConsoleHostUserInterfaceStrings.PromptForChoiceHelp,
-                fg,
-                bg,
-                ref lineLen,
-                lineLenMax);
-            if (shouldEmulateForMultipleChoiceSelection)
-            {
-                WriteLineToConsole();
+                prompts.Add(PSStyle.Decorate(choice, color));
             }
 
-            string defaultPrompt = string.Empty;
+            prompts.Add(PSStyle.Decorate(ConsoleHostUserInterfaceStrings.PromptForChoiceHelp,
+                PSStyle.Instance.Prompt.ChoiceHelp) + " ");
+
+            WriteToConsole(shouldEmulateForMultipleChoiceSelection
+                ? string.Join(Environment.NewLine, prompts)
+                : JoinWrappedToCurrentWindowWidth("  ", prompts),
+                transcribeResult: true);
```

Comment:
Eh, looks too expensive. If I understand correctly we add color then parse string again to calculate real length... looks weird :-(

Reply:
That's roughly what the old code already did. And this is code is only ever used for user prompts, so does it really matter whether we render it in 0.1ms or 0.3ms, if the user takes couple seconds to read, understand and press a button? I get optimizing hot paths, but this is not one.

Reply:
@iSazonov Could you please reply or mark the comment as resolved?

Reply:
I think an alternative would add complexity in a different way.